src/sagemaker/estimator.py

```diff
@@ -442,6 +444,21 @@ def _prepare_for_training(self, job_name=None):
         self._prepare_debugger_for_training()
         self._prepare_profiler_for_training()
 
+    def _validate_framework_rule_compatibility(self, rule):
+        if rule.rule_type != RuleType.DEBUGGER_BUILTIN:
+            # We only validate DEBUGGER BUILTIN RULES for framework compatibility
+            # Since, profiler rules are framework agnostic
```

Comment:
nit: let's add `debugger` to the function name so that it's clear this is only meant to be called for debugger rules.

Reply:
also, just put `return` since the return value of this function isn't used for anything.

Reply:
Done.